magicctapipe/scripts/lst1_magic/semi_automatic_scripts/setting_up_config_and_dir.py

```diff
@@ -0,0 +1,740 @@
+"""
+This script facilitates the usage of
+"magic_calib_to_dl1.py". This script is more like a
+"manager" that organizes the analysis process by:
+1) Creating the necessary directories and subdirectories.
+2) Generating all the bash script files that convert the
+MAGIC files from DL0 to DL1.
+3) Launching these jobs in the IT container.
+
+Notice that in this stage we only use MAGIC data.
+No LST data is used here.
+
+Standard usage:
+$ python setting_up_config_and_dir.py (-c config_file.yaml)
+"""
+import argparse
+import glob
+import logging
+import os
+
+# import time
+from pathlib import Path
+
+import joblib
+import numpy as np
+import yaml
+
+from magicctapipe import __version__
+from magicctapipe.io import resource_file
+
+__all__ = [
+    "config_file_gen",
+    "lists_and_bash_generator",
+    "lists_and_bash_gen_MAGIC",
+    "directories_generator",
+]
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.StreamHandler())
+logger.setLevel(logging.INFO)
+
+
+def config_file_gen(ids, target_dir, noise_value, NSB_match, source_name):
+
+    """
+    Here we create the configuration file needed for transforming DL0 into DL1
+
+    Parameters
+    ----------
+    ids : list
+        Telescope IDs
+    target_dir : path
+        Directory to store the results
+    noise_value : list
```

Comment:
I do not understand this part. The script is meant only for calib to DL1 files of the data, so why do you have the part of adding the noise, that is only done for MCs?

Reply:
To 'update' the standard config (written by this function) with the values we want to use to process the MCs (extra_noise_in_dim_pixels...)